src/auth/guards/roles.guard.ts

```diff
@@ -0,0 +1,35 @@
+import { from, Observable } from 'rxjs';
+import { Reflector } from '@nestjs/core';
+import { filter, map, pluck } from 'rxjs/operators';
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+
+import { User } from '../../users/entities/user.entity';
+import { UsersService } from '../../users/users.service';
+import { UserRole } from '../../common/enums/user-role.enum';
+import { ROLES_KEY } from '../decorators/roles.decorator';
+
+@Injectable()
+export class RolesGuard implements CanActivate {
+  constructor(
+    private readonly reflector: Reflector,
+    private readonly userService: UsersService,
+  ) {}
+
+  canActivate(context: ExecutionContext): boolean | Observable<boolean> {
+    const requiredRoles: UserRole[] = this.reflector.getAllAndOverride(
+      ROLES_KEY,
+      [context.getHandler(), context.getClass()],
+    );
+    if (!requiredRoles || requiredRoles.length < 1) {
```

Comment:
what if requiredRoles = undefined?
```
requiredRoles.length // cannot read property length of undefined
```

Reply:
@tar-aldev if requiredRoles is undefined then should return true. That means if we ha't specific roles for the route, the check this guard always return true